libbeat/docs/https.asciidoc

```diff
@@ -10,66 +10,152 @@
-use environment variables.
+in environment variables.
 endif::[]
+--
 
-{beatname_uc} verifies the validity of the server certificates and only accepts trusted
-certificates. Creating a correct SSL/TLS infrastructure is outside the scope of
-this document.
-
-By default {beatname_uc} uses the list of trusted certificate authorities from the
-operating system where {beatname_uc} is running. You can configure {beatname_uc} to use a specific list of
-CA certificates instead of the list from the OS. You can also configure it to use client authentication
-by specifying the certificate and key to use when the server requires the {beatname_uc} to authenticate. Here is an example
-configuration:
+* To use token-based *API key authentication*, specify the `api_key` under `output.elasticsearch`.
+For example:
++
+--
+["source","yaml",subs="attributes,callouts"]
+----------------------------------------------------------------------
+output.elasticsearch:
+  hosts: ["https://localhost:9200"]
+  api_key: "KnR6yE41RrSowb0kQ0HWoA" <1>
+----------------------------------------------------------------------
+<1> This API key must have the privileges required to publish events to {es}.
+To create an API key like this, see <<beats-api-keys>>.
+--
 
+[[beats-tls]]
+* To use *Public Key Infrastructure (PKI) certificates* to authenticate users,
+specify the `certificate` and `key` settings under `output.elasticsearch`.
+For example:
++
+--
+["source","yaml",subs="attributes,callouts"]
+----------------------------------------------------------------------
+output.elasticsearch:
+  hosts: ["localhost:9200"]
+  ssl.certificate: "/etc/pki/client/cert.pem" <1>
+  ssl.key: "/etc/pki/client/cert.key" <2>
+----------------------------------------------------------------------
+<1> The path to the certificate for SSL client authentication
+<2> The client certificate key
+--
++
+These settings assume that the
+distinguished name (DN) in the certificate is mapped to the appropriate roles in
+the `role_mapping.yml` file on each node in the {es} cluster. For more
+information, see {ref}/mapping-roles.html#mapping-roles-file[Using role
+mapping files].
++
+By default, {beatname_uc} uses the list of trusted certificate authorities (CA) from the
+operating system where {beatname_uc} is running. If the certificate authority that signed your node certificates
+is not in the host system's trusted certificate authorities list, you need
+to add the path to the `.pem` file that contains your CA's certificate to the
+{beatname_uc} configuration. This will configure {beatname_uc} to use a specific list of
+CA certificates instead of the default list from the OS.
++
+Here is an example configuration:
++
+--
 ["source","yaml",subs="attributes,callouts"]
 ----------------------------------------------------------------------
 output.elasticsearch:
-  username: {beatname_lc}
-  password: verysecret
-  protocol: https
-  hosts: ["elasticsearch.example.com:9200"]
+  hosts: ["localhost:9200"]
```

Comment:
same comment as before about including https. I guess it might be the default now?